force-app/main/default/classes/PMT_ValidationService.cls

```diff
@@ -0,0 +1,211 @@
+        if (this.isEnforceAccountingDataConsistency) {
+            enforceAccountingDataConsistency(payment, errorCollection);
+        }
+
+        if (payment.npe01__Paid__c == true && payment.npe01__Written_Off__c == true) {
+            errorCollection.addError(System.Label.pmtWriteOffPaidPayment);
+        }
+    }
+
+    /*******************************************************************************************************
+    * @description If the custom setting to enforce account rules is enabled, it will validate that paid payments
+    * have a payment date and unpaid payments have a scheduled date.
+    * @param payment the Payment from trigger new.
+    * @param errorCollection Collection of error
+    ********************************************************************************************************/
+    private void enforceAccountingDataConsistency(npe01__OppPayment__c payment, ErrorRecord errorCollection) {
+        if ((payment.npe01__Paid__c == true || payment.npe01__Written_Off__c == true) && payment.npe01__Payment_Date__c == null) {
+
+            if (payment.npe01__Paid__c == true) {
+                errorCollection.addError(System.Label.pmtPaidPaymentDateRequired);
+
+            } else {
+                errorCollection.addError(System.Label.pmtWrittenOffPaymentDateRequired);
+
+            }
+        } else if (payment.npe01__Paid__c == false && payment.npe01__Written_Off__c == false && payment.npe01__Scheduled_Date__c == null) {
+            errorCollection.addError(System.Label.pmtUnpaidPaymentScheduledDateRequired);
+        }
+    }
+
+    private void validateRefunds(npe01__OppPayment__c payment, ErrorRecord errorCollection) {
+        if (payment.DebitType__c != PMT_RefundService.FULL_REFUND) {
+            return;
+        }
+
+        PMT_RefundService.RefundInfo refundInfo = originalPaymentWithRefunds.get(payment.OriginalPayment__c);
+
+        if (payment.OriginalPayment__c == null || refundInfo == null || refundInfo.originalPayment == null) {
+            errorCollection.addError(System.Label.pmtOriginalPaymentNotFound);
+            return;
+        }
+
+        if (refundInfo.originalPayment.npe01__Paid__c != true) {
+            errorCollection.addError(System.Label.pmtOriginalPaymentNotPaid);
+        }
+
+        if (refundInfo.originalPayment.DebitType__c == PMT_RefundService.FULL_REFUND) {
+            errorCollection.addError(System.Label.pmtPaymentNotRefundable);
+        }
+
+        if (refundInfo.refundableAmount + payment.npe01__Payment_Amount__c != 0) {
+            errorCollection.addError(System.Label.pmtRefundAmountInvalid);
+        }
+
+        refundInfo.refundableAmount += payment.npe01__Payment_Amount__c;
+    }
+
+    private void validateElevatePayments(npe01__OppPayment__c payment, npe01__OppPayment__c oldPayment, ErrorRecord errorCollection) {
+        if (String.isBlank(payment.Elevate_Payment_ID__c) || !config.isIntegrationEnabled() || config.hasIntegrationPermissions()) {
```

Comment:
Aren't we supposed to check if the org does not have `integrationPermissions`? so `!config.hasIntegrationPermissions()`